src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            addFirstNode(value);
+        } else {
+            addNodeToEnd(value);
+        }
     }
```

Comment:
When creating a new `Node<T>`, pass all the fields to the `Node` constructor. It looks like the `addFirstNode` and `addNodeToEnd` methods are not doing this properly.

Reply:
I don't create here Node!